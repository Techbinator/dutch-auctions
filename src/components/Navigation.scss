@import "../config/scss/variables.scss";
nav {
  display: flex;
  margin-top: 10px;
  min-width: 300px;
  a {
    text-decoration: none;
    font-size: 1em;
    padding-left: 5%;
    white-space: nowrap;
    color: #000;
    &.active,
    &:hover {
      color: $hover;
    }
    .svg-inline--fa {
      margin-right: 5px;
    }
  }
}
@media only screen and (max-width: 768px) {
  nav {
    flex-direction: column;
    width: 100%;
    text-align: center;
    display: none;
    a {
      padding: 5px 0;
      font-size: 1.1em;
      border-top: 1px solid #ccc;
    }
    &.active {
      display: flex;
    }
  }
}
