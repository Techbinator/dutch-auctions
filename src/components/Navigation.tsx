import React from "react";
import { NavLink } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faSignOutAlt,
  faUserCircle,
  faListAlt
} from "@fortawesome/free-solid-svg-icons";

interface INavigation {
  showMenu: boolean;
}
export default ({ showMenu }: INavigation) => (
  <nav className={showMenu ? "active" : ""}>
    <NavLink exact to="/auctions">
      <FontAwesomeIcon icon={faListAlt} />
      Auctions
    </NavLink>
    <NavLink exact to="/my_auctions">
      <FontAwesomeIcon icon={faUserCircle} />
      My auctions
    </NavLink>
    <NavLink exact to="/logout">
      <FontAwesomeIcon icon={faSignOutAlt} />
      Log out
    </NavLink>
  </nav>
);
