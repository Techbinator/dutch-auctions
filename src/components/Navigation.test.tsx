import React from "react";
import Navigation from "./Navigation";
import { shallow } from "enzyme";

const onLogoutClick = jest.fn();
describe("Navigation component", () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = shallow(
      <Navigation isLogedIn showMenu onLogOutClick={onLogoutClick} />
    );
  });
  it("should render snapshot with loged in urls in header", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render snapshot with loged out urls in header", () => {
    const onLogoutClick = jest.fn();
    const wrapper = shallow(
      <Navigation showMenu={false} onLogOutClick={onLogoutClick} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should call log out function on link click", () => {
    wrapper.find(".logout").simulate("click");
    expect(onLogoutClick).toBeCalled();
  });
});
