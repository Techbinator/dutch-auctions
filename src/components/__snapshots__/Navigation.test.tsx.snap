// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should render snapshot with loged in urls in header 1`] = `
<nav
  className="active"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    exact={true}
    to="/auctions"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f022",
            "M464 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zM128 120c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm0 96c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm0 96c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm288-136v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12zm0 96v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12zm0 96v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12z",
          ],
          "iconName": "list-alt",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    Auctions
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    exact={true}
    to="/my_auctions"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            496,
            512,
            Array [],
            "f2bd",
            "M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z",
          ],
          "iconName": "user-circle",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    My auctions
  </NavLink>
  <a
    className="logout"
    href="#"
    onClick={[MockFunction]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f2f5",
            "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
          ],
          "iconName": "sign-out-alt",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    Log out
  </a>
</nav>
`;

exports[`Navigation component should render snapshot with loged out urls in header 1`] = `
<nav
  className=""
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    exact={true}
    to="/login"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f2f6",
            "M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z",
          ],
          "iconName": "sign-in-alt",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    Log In
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    exact={true}
    to="/register"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f25d",
            "M285.363 207.475c0 18.6-9.831 28.431-28.431 28.431h-29.876v-56.14h23.378c28.668 0 34.929 8.773 34.929 27.709zM504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM363.411 360.414c-46.729-84.825-43.299-78.636-44.702-80.98 23.432-15.172 37.945-42.979 37.945-74.486 0-54.244-31.5-89.252-105.498-89.252h-70.667c-13.255 0-24 10.745-24 24V372c0 13.255 10.745 24 24 24h22.567c13.255 0 24-10.745 24-24v-71.663h25.556l44.129 82.937a24.001 24.001 0 0 0 21.188 12.727h24.464c18.261-.001 29.829-19.591 21.018-35.587z",
          ],
          "iconName": "registered",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    Register
  </NavLink>
</nav>
`;
